usaspending_api/references/v2/views/cfda.py

```diff
@@ -57,7 +64,13 @@ def _populate_cfdas_if_needed(self):
             CFDA_DICTIONARY = response
 
     def _request_from_grants_api(self):
-        return post(
-            "https://www.grants.gov/grantsws/rest/opportunities/search/cfda/totals",
+        cfda_response = post(
+            CFDA_URL,
             headers={"Authorization": f"APIKEY={settings.GRANTS_API_KEY}"},
-        ).json()["cfdas"]
+        )
+        if cfda_response.status_code == status.HTTP_503_SERVICE_UNAVAILABLE:
+            raise ServiceUnavailable(f"{CFDA_URL} not available (status 503)")
+        if cfda_response.json()["errorMsgs"] != []:
+            raise InternalServerError(f"Error returned by {CFDA_URL}: {cfda_response.json()['errorMsgs']}")
```

Comment:
Again, I discourage returning a 500. Raised Django API exceptions should be specific